compiler/src/org.graalvm.compiler.truffle.compiler.benchmark/src/org/graalvm/compiler/truffle/compiler/benchmark/pelang/PELangExpressionNode.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2018, 2018, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package org.graalvm.compiler.truffle.compiler.benchmark.pelang;
+
+import com.oracle.truffle.api.dsl.TypeSystemReference;
+import com.oracle.truffle.api.frame.VirtualFrame;
+import com.oracle.truffle.api.nodes.Node;
+import com.oracle.truffle.api.nodes.UnexpectedResultException;
+
+@TypeSystemReference(PELangTypes.class)
+public abstract class PELangExpressionNode extends Node {
+
+    public abstract Object executeGeneric(VirtualFrame frame);
+
+    public long executeLong(VirtualFrame frame) throws UnexpectedResultException {
+        return PELangTypesGen.expectLong(executeGeneric(frame));
```

Comment:
Doesn't look its worth using a typesystem to generate these to expectLong methods. I'd just remove the typesystem and inline its implementation here.

Reply:
Yeah you are right, I will definitively do this.